encode/src/main/scala/org/broadinstitute/monster/etl/encode/extract/EncodeExtractions.scala

```diff
@@ -0,0 +1,209 @@
+import com.spotify.scio.values.SCollection
+import com.squareup.okhttp.{Callback, OkHttpClient, Request, Response}
+import org.apache.beam.sdk.coders.{KvCoder, StringUtf8Coder}
+
+import scala.collection.JavaConverters._
+import scala.util.Success
+
+class EncodeSearch(encodeApiName: String)
+    extends AsyncLookupDoFn[List[(String, String)], String, AsyncHttpClient](100) {
+
+  override def asyncLookup(
+    client: AsyncHttpClient,
+    params: List[(String, String)]
+  ): ListenableFuture[String] = {
+    val paramsString = List(
+      "frame=object",
+      "status=released",
+      s"type=$encodeApiName",
+      "limit=all",
+      "format=json"
+    ) ::: params.map {
+      case (key, value) =>
+        s"$key=$value"
+    }
+
+    Futures.transform(
+      client.get(
+        "https://www.encodeproject.org/search/?" + paramsString.mkString(sep = "&")
+      ),
+      (input: Option[String]) => input.get,
+      MoreExecutors.directExecutor
+    )
+  }
+
+  override protected def newClient(): AsyncHttpClient = new AsyncHttpClient
+}
+
+class AsyncHttpClient {
+
+  def get(url: String): ListenableFuture[Option[String]] = {
+    val request = new Request.Builder()
+      .url(url)
+      .get
+      .build
+
+    val result: SettableFuture[Option[String]] = SettableFuture.create()
+    EncodeExtractions.client
+      .newCall(request)
+      .enqueue(new Callback() {
+        def onFailure(request: Request, exception: IOException): Unit = {
+          result.setException(exception)
+          ()
+        }
+
+        def onResponse(response: Response): Unit = {
+          if (response.isSuccessful) {
+            result.set(Option(response.body.string))
+            ()
+          } else {
+            result.set(Option("Default Value"))
+            ()
+          }
+        }
+      })
+    result
+  }
+}
+
+/** Ingest step responsible for pulling raw metadata for a specific entity type from the ENCODE API. */
+object EncodeExtractions {
+
+  val client = new OkHttpClient()
+
+  implicit def coderTry: Coder[Try[String]] = Coder.kryo[Try[String]]
+
+  /**
+    * Add [("frame", "object"), ("status", "released")] to the parameters
+    * and then call the ENCODE search client API using the query id/search parameters.
+    *
+    * @param entryName the name of the entity type to be displayed as a step within the pipeline
+    * @param encodeApiName the entity type that will be queried
+    **/
+  def extractEntities(
+    entryName: String,
+    encodeApiName: String
+  ): SCollection[List[(String, String)]] => SCollection[JsonObject] =
+    _.transform(s"Extract $entryName entities") {
+      _.applyKvTransform(ParDo.of(new EncodeSearch(encodeApiName))).flatMap { kv =>
+        kv.getValue.asScala match {
+          case Success(value) =>
+            io.circe.parser
+              .parse(value)
+              .flatMap { json =>
+                json.as[Vector[JsonObject]]
+              }
+              .right
+              .get
```

Comment:
Instead of `.right.get`, we should do `.fold(throw _, identity)`. It'll still explode, but it'll show a better error trace.